src/Compilers/CSharp/Portable/Symbols/Source/SourceEventDefinitionAccessorSymbol.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols;
+
+/// <summary>
+/// Accessor of a <see cref="SourceFieldLikeEventSymbol"/> which is a partial definition.
+/// </summary>
+internal sealed class SourceEventDefinitionAccessorSymbol : SourceEventAccessorSymbol
+{
+    internal SourceEventDefinitionAccessorSymbol(
+        SourceFieldLikeEventSymbol ev,
+        bool isAdder,
+        BindingDiagnosticBag diagnostics)
+        : base(
+            @event: ev,
+            syntaxReference: ev.SyntaxReference,
+            location: ev.Location,
+            explicitlyImplementedEventOpt: null,
+            aliasQualifierOpt: null,
+            isAdder: isAdder,
+            isIterator: false,
+            isNullableAnalysisEnabled: ev.DeclaringCompilation.IsNullableAnalysisEnabledIn(ev.CSharpSyntaxNode),
+            isExpressionBodied: false)
+    {
+        CheckFeatureAvailabilityAndRuntimeSupport(ev.CSharpSyntaxNode, ev.Location, hasBody: false, diagnostics: diagnostics);
+    }
+
+    public override Accessibility DeclaredAccessibility => AssociatedEvent.DeclaredAccessibility;
+
+    public override bool IsImplicitlyDeclared => true;
+
+    internal override bool GenerateDebugInfo => true;
```

Comment:
>GenerateDebugInfo => true [](http://example.com/codeflow?start=27&length=25)

Is this correct? If we are generating code based on this accessor symbol, what if the event is marked `extern`?

Reply:
Looks like this is never accessed on partial definitions, it is only accessed on partial implementations by the emit layer. So in theory this could be changed to throw unreachable exception if you think that would be better.

Generally `true` seems correct here - this accessor symbol represents a user-defined method and GenerateDebugInfo controls emit of sequence points and Debug-style IL which we want I think for all user-defined methods. 

Note that property accessors also return constant `true` for this.

Regarding `extern`, those have empty bodies, so GenerateDebugInfo should have no effect there.